apps/communications/dialer/test/marionette/keypad_test.js

```diff
@@ -11,87 +12,89 @@ marionette('Dialer > Keypad', function() {
-    var sequence = [one, two, three];
-    var length = sequence.length;
-    for (var i = 0, digit; (digit = sequence[i]); i = (i+1) % length) {
-      keypadTap(digit);
-      numberTyped += digit.getAttribute('data-value');
-      if (number.getAttribute('value').indexOf('\u2026') === 0) {
-        return numberTyped;
+    function typeLongNumber() {
+      var number = subject.client.findElement(selectors.phoneNumber);
+      var numberTyped = '';
+
+      var one = subject.client.findElement(selectors.one);
+      var two = subject.client.findElement(selectors.two);
+      var three = subject.client.findElement(selectors.three);
+      var sequence = [one, two, three];
+      var length = sequence.length;
+      for (var i = 0, digit; (digit = sequence[i]); i = (i+1) % length) {
+        keypadTap(digit);
+        numberTyped += digit.getAttribute('data-value');
+        if (number.getAttribute('value').indexOf('\u2026') === 0) {
+          return numberTyped;
+        }
       }
     }
-  }
-
-  // Avoid "all the same number" patterns like 111111... because, under
-  // ellipsis scenaries, the value of the input won't change and the test
-  // will timeout.
-  function keypadTap(elem, longPress) {
-    var number = subject.client.findElement(selectors.phoneNumber);
-    var previousValue = number.getAttribute('value');
-
-    if (longPress) {
-      actions.longPress(elem, 1).perform();
-    } else {
-      actions.tap(elem).perform();
-    }
 
-    client.waitFor(function () {
-      return number.getAttribute('value') !== previousValue;
-    });
-  }
+    // Avoid "all the same number" patterns like 111111... because, under
+    // ellipsis scenaries, the value of the input won't change and the test
+    // will timeout.
+    function keypadTap(elem, longPress) {
+      var number = subject.client.findElement(selectors.phoneNumber);
+      var previousValue = number.getAttribute('value');
+
+      if (longPress) {
+        actions.longPress(elem, 1).perform();
+      } else {
+        actions.tap(elem).perform();
+      }
 
-  function loadSuggestionDOM() {
-    typeNumber();
+      client.waitFor(function () {
+        return number.getAttribute('value') !== previousValue;
+      });
+    }
 
-    var del = subject.client.findElement(selectors.del);
-    actions.longPress(del, 1).perform();
+    function loadSuggestionDOM() {
+      typeNumber();
 
-    var number = subject.client.findElement(selectors.phoneNumber);
-    client.waitFor(function() {
-      return (number.getAttribute('value') === '');
-    });
-  }
+      var del = subject.client.findElement(selectors.del);
+      actions.longPress(del, 1).perform();
 
-  test('Add a contact from dialer', function() {
+      var number = subject.client.findElement(selectors.phoneNumber);
+      client.waitFor(function() {
+        return (number.getAttribute('value') === '');
+      });
+    }
+
+    test('Add a new contact from dialer', function() {
```

Comment:
I thought that also, but the set up is different, I mean, for this test it does not matter to create a contact first and I do that for the other test in the set up for the test suite. Do not know if there is another way to initialize only one test with a set up. Otherwise a contact is created for this test although it is not needed...